habitat-lab/habitat/tasks/rearrange/rearrange_grasp_manager.py

```diff
@@ -307,6 +314,11 @@ def snap_to_obj(
         :param force: Will kinematically snap the object to the robot's end-effector, even if
             the object is already in the grasped state.
         """
+
+        if snap_obj_id in self._did_place:
```

Comment:
this should use the `self._block_double_pick` flag too right?

Reply:
Uh oh you're right. I just fixed.